hack/proto-codegen.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeCarrier Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/usr/bin/env bash
+
+set -eu
+
+# As per official docs
+# https://grpc-ecosystem.github.io/grpc-gateway/docs/usage.html
+
+# Add protoc and protoc-gen-go tools to PATH
+export PATH=${PWD}/bin:$PATH
+PROJECT=$PWD
+GOPATH=$(go env GOPATH)
+
+PBUFS=(
+  pkg/apis/apiserver/v1alpha1
+)
+
+# change into each protobuf directory
+for pkg in ${PBUFS} ; do
+  abs_path=${PROJECT}/${pkg}
+  echo Generating from '*.proto' in $abs_path
+  protoc \
+    --go_out=plugins=grpc:${abs_path}  \
+    --grpc-gateway_out=logtostderr=true:${abs_path} \
+    --swagger_out=logtostderr=true:${abs_path} \
+    -I${PROJECT}/bin/protoc-bin/include \
+    -I${GOPATH}/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.14.3/third_party/googleapis \
+    -I/usr/include \
```

Comment:
I like system wide protoc installation more.

Reply:
but then why do you need `-I${PROJECT}/bin/protoc-bin/include \`?

Reply:
@nmiculinic why was this marked as resolved?

Reply:
see `Makefile` ; this is used within the `quay.io/kubecarrier/test` container since it's the easier way for its replicability. Inside the container the protoc is installed with the proper version under `/usr/include`

Reply:
also, there's a comment at file's beginning.